packages/ui/src/Toast.tsx

```diff
@@ -53,58 +79,154 @@ export function useToast(): {
-      )}
-    </Box>
+      <View
+        style={{
+          display: "flex",
+          flexShrink: 1,
+          flexDirection: "row",
+          minWidth: 150,
+          paddingTop: theme.spacing.xs as any,
+          paddingBottom: theme.spacing.xs as any,
+          paddingRight: theme.spacing.sm as any,
+          alignItems: "center",
+          gap: 10,
+          borderRadius: theme.radius.default as any,
+          backgroundColor: theme.surface[color],
+          alignSelf: "flex-start",
+          minHeight: size === "lg" ? 32 : undefined,
+          maxWidth: "100%", // Ensure the content does not overflow
+        }}
+      >
+        <View
+          style={{
+            paddingLeft: 8,
+            paddingRight: 8,
+            display: "flex",
+            flexDirection: "row",
+            alignItems: "center",
+            gap: 12,
+            maxWidth: "100%",
+            flexShrink: 1, // Ensure the content can shrink properly
+            flexGrow: 1,
+          }}
+        >
+          <View
+            style={{
+              display: "flex",
+              flexDirection: "row",
+              paddingTop: size === "lg" ? 8 : 0,
+              paddingBottom: size === "lg" ? 8 : 0,
+              paddingLeft: size === "lg" ? 4 : 0,
+              paddingRight: size === "lg" ? 4 : 0,
+              alignItems: size === "lg" ? "center" : undefined,
+              alignSelf: size === "lg" ? "stretch" : undefined,
+              borderTopLeftRadius: theme.radius.default as any,
+              borderBottomLeftRadius: theme.radius.default as any,
+            }}
+          >
+            <Icon color={textColor} iconName={iconName} size={size === "lg" ? "2xl" : "md"} />
+          </View>
+          <View
+            style={{
+              display: "flex",
+              paddingTop: 8,
+              paddingBottom: 8,
+              flexDirection: "column",
+              justifyContent: "center",
+              alignItems: "flex-start",
+              gap: 2,
+              alignSelf: "stretch",
+              flexWrap: "wrap",
+              flexShrink: 1, // Ensure the content can shrink properly
+            }}
+          >
+            {Boolean(size === "lg") ? (
+              <Heading color={textColor} size="sm">
+                {title}
+              </Heading>
+            ) : (
+              <Text bold color={textColor} size="md">
+                {title}
+              </Text>
+            )}
+            {Boolean(size === "lg" && subtitle) && (
+              <Text color={textColor} size="sm">
+                {subtitle}
+              </Text>
+            )}
+          </View>
+        </View>
+        <Pressable
+          accessibilityRole="button"
+          style={{
+            display: "flex",
+            alignItems: "center",
+            alignSelf: "center",
+            gap: 12,
+            marginLeft: 10,
+            padding: size === "lg" ? 8 : 0,
+          }}
+          onPress={onDismiss}
```

Comment:
In demo, pressing did not dismiss the toast sooner

Reply:
Ahh, it's because there shouldn't be a dismiss unless it's persistent.